Pidgin.Tests/ParserTestBase.cs

```diff
@@ -19,5 +19,13 @@ public class ParserTestBase
             Assert.Equal(errorPos, result.Error.ErrorPos);
             Assert.Equal(consumedInput, result.ConsumedInput);
         }
+
+        protected void AssertUnexpectedFailure<TToken, T>(Result<TToken, T> result, Maybe<TToken> unexpected, SourcePos errorPos, bool consumedInput)
+        {
+            Assert.False(result.Success);
+            Assert.Equal(unexpected, result.Error.Unexpected);
+            Assert.Equal(errorPos, result.Error.ErrorPos);
+            Assert.Equal(consumedInput, result.ConsumedInput);
+        }
```

Comment:
Yeah, that's what I was thinking. (Don't do that work yourself, it'll be super boring 😜)